### ISSUE TITLE
Produce storage info function

### ISSUE THREAD

**Issue by Scrubsauce** [mindaptiv/Saul#54]

**Comment by Scrubsauce** [comment]
StatFs class reference: http://developer.android.com/reference/android/os/StatFs.html#getBlockSizeLong()

**Comment by Scrubsauce** [comment]
"External" vs Mountable: http://stackoverflow.com/questions/12087510/check-if-the-sdcard-is-present-boolean-is-always-true

**Comment by Scrubsauce** [comment]
Measure just card?
http://stackoverflow.com/questions/16267116/how-to-get-actual-size-of-mounted-sd-card-in-android

**Comment by Scrubsauce** [comment]
Storage Options ref:
http://developer.android.com/guide/topics/data/data-storage.html#filesExternal

**Comment by Scrubsauce** [comment]
Environment class ref: https://developer.android.com/reference/android/os/Environment.html#getExternalStorageDirectory()

**Comment by Scrubsauce** [comment]
Android's got a bit of a troubled history with how it handles SD cards: http://stackoverflow.com/questions/5694933/find-an-external-sd-card-location

**Comment by Scrubsauce** [comment]
Good write up explains how it all goes together:
http://possiblemobile.com/2014/03/android-external-storage/

**Comment by Scrubsauce** [comment]
Possible way to find other secondary storage part 1 and 2
http://stackoverflow.com/questions/11281010/how-can-i-get-external-sd-card-path-for-android-4-0
http://stackoverflow.com/questions/11281010/how-can-i-get-external-sd-card-path-for-android-4-0/27197248#27197248

**Comment by Scrubsauce** [comment]
TL;DR primary external storage is no longer always external and in many cases (Samsung, etc.) instead points to the internal flash disk of a device.  Also, permissions are wonky at times for where an app can actually write, so not all additional storage (sd cards, flash drives attached via cable, etc.) is always available in full capacity.

**Comment by Scrubsauce** [comment]
Preliminary producer in place for testing the primary storage location: 
ac2fb285808ce3b3abb277b22315fd39e0e73ab9

**Comment by Scrubsauce** [comment]
Pay attention to Lozenko's method
http://stackoverflow.com/questions/11281010/how-can-i-get-external-sd-card-path-for-android-4-0/27197248#27197248

**Comment by Scrubsauce** [comment]
Secondary Storage (Warning: cable related storage not tested!!!) support as of: 
c682e15f5525fdaf995a13467519d60ef49a68d1

**Comment by Scrubsauce** [comment]
Device object creation for supported storage devices as of:
54d24df6bf14f1905b23d7fa9b4dc29a4508eee4

**Comment by Scrubsauce** [comment]
Storage object creation for support devices as of:
300b6065e6165193c6d604be7e4d354f75d72dcf

**Comment by Scrubsauce** [comment]
OTG cables are ordered, should be ~mid-May.

**Comment by Scrubsauce** [comment]
Devices created with this function now have their deviceType "enum" set to 17 (storage device mapping) as of 272977b3f73aa4acb1e13e086c984f9564e5f625

**Comment by Scrubsauce** [comment]
Storage objects now denote if they are emulated with the isEmulated field as of 610e8540bb0a7d797bd7c61ea6807ff06cb4592a

**Comment by Scrubsauce** [comment]
Now logging contents of "storages" linked list w/in test logging method:
e708cd2122576edec6a2ff58aca3533558b93dbd

**Comment by Scrubsauce** [comment]
Fixed storageIndex values logging incorrectly as of https://github.com/mindaptiv/Saul/commit/d009db16c236785f8fe57e3ae39e89049c732766

**Comment by Scrubsauce** [comment]
Temporary hold on storage devices until OTG cables arrive.

**Comment by Scrubsauce** [comment]
Basic functionality complete, split into #75 with other producers for OTG testing.
